src/Shared/CertificateGeneration/NssCertificateDatabase.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Tools.Internal;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Certificates.Generation
+{
+    internal class NssCertificateDatabase : CertificateStore
+    {
+        public string DatabasePath { get; }
+
+        public NssCertificateDatabase(string name, string path) :
+            base(name)
+        {
+            DatabasePath = path;
+        }
+
+        public override bool CheckDependencies(IReporter? reporter)
+        {
+            return CheckProgramDependency("certutil", reporter);
+        }
+
+        public override bool TryInstallCertificate(string name, PemCertificateFile pemFile, IReporter? reporter, bool isInteractive)
+        {
+            var result = ProcessRunner.Run(new() {
+                Command = { "certutil", "-d", DatabasePath, "-A", "-t", "C,,", "-n", name, "-i", pemFile.FilePath },
+                ThrowOnFailure = false });
+            if (!result.IsSuccess)
+            {
+                reporter?.Error($"Failed to install certificate using command '{result.CommandLine}': {result.StandardError}");
+            }
+            return result.IsSuccess;
+        }
+
+        public override bool HasCertificate(string name, X509Certificate2 certificate)
+        {
+            ProcessRunResult runResult = ProcessRunner.Run(new()
+            {
+                Command = { "certutil", "-d", DatabasePath, "-L", "-n", name, "-a" },
+                ReadStandardOutput = true,
+                ThrowOnFailure = false
+            });
+            if (runResult.IsSuccess)
+            {
+                runResult.StandardOutput!.Replace("\r\n", "\n");
+                const string BeginCertificate = "-----BEGIN CERTIFICATE-----";
+                var pemCertificates = runResult.StandardOutput.ToString()
+                    .Split(BeginCertificate, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var pem in pemCertificates)
+                {
+                    X509Certificate2 cert = X509Certificate2.CreateFromPem(BeginCertificate + "\n" + pem);
+                    if (cert.Equals(certificate))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
```

Comment:
Instead of doing this, can we make the certificate name be `aspnetcore-https-{sha256}`? with two goals in mind:
1. We can exactly identify each certificate by querying for the name.
2. We can clean all certificates by listing those that start with aspnetcore-https.

Reply:
> can we make the certificate name be aspnetcore-https-{sha256}

 I've used `aspnet-{Thumbprint}`. Is that ok?

> We can clean all certificates by listing those that start with aspnetcore-https.

We don't want to delete other user's certificates.
For now, it deletes specific certificates.

Reply:
aspnetcore-{thumbprint} sounds good.

> We don't want to delete other user's certificates.

`dotnet dev-certs https --clean` needs to remove all certificates that we have created. I'm fine if we `claim` that namespace and remove any certificate that has been saved as `aspnet-{thumbprint}`. If you are not happy with that, then we need to load the certificates and check for the presence of the specific OID our tool sets up on the certificate.  There is a function in certificate manager to check if the certificate is an asp.net core https certificate (And if someone has decided to create their own certificate and use our OID (which we explicitly discourage people from) then tough luck for them)

Reply:
Should `dotnet dev-certs https --clean` remove certificates that were generated and trusted by other users?

Reply:
No, we should only clean the certificates we create (that's why we use prefix/thumbprint so we can tell which are ours)